internal/http/services/appprovider/appprovider.go

```diff
@@ -146,16 +142,80 @@ func (s *svc) handleNew(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	target := r.URL.Query().Get("filename")
-	if target == "" {
+	rootID := r.URL.Query().Get("file_root_id")
```

Comment:
As this is now a new required parameter, can you reference the corresponding PR in web to pass it?
Also, a cosmetic comment: would it make more sense to call the parameter `root_file_id`?

Reply:
Yes, even better!